extensions-core/lookups-cached-global/src/main/java/io/druid/server/lookup/namespace/cache/NamespaceExtractionCacheManager.java

```diff
@@ -289,10 +280,7 @@ public void onFailure(Throwable t)
           }
         }
     );
-    if (!future.isDone()
-        && !future.cancel(true)) { // Interrupt to make sure we don't pollute stuff after we've already cleaned up
-      throw new ISE("Future for namespace [%s] was not able to be canceled", implDatum.name);
-    }
+    future.cancel(true);
```

Comment:
@fjy the future could be actually cancelled after `isDone()` call and before `cancel()`. Also we don't really care here if the future is already cancelled/completed or not, we just want to cancel if it still running.